DYDX-Bots/Insolvent_Contract/src/agent.spec.ts

```diff
@@ -0,0 +1,123 @@
+import { FindingType, FindingSeverity, Finding, HandleBlock, BlockEvent } from "forta-agent";
+import { createAddress, TestBlockEvent, MockEthersProvider } from "forta-agent-tools/lib/tests";
+import { BigNumber } from "ethers";
+import { provideHandleBlock } from "./agent";
+import NetworkData from "./network";
+import BalanceFetcher from "./balance.fetcher";
+import { IMPLEMENTATION_IFACE } from "./utils";
+
+const testBlockNumbers: number[] = [2, 42, 92, 360];
+const testThreshold: BigNumber = BigNumber.from("3000000000000000000"); // 3
+const testBorrowerDebtBalance: BigNumber[] = [
+  BigNumber.from("5000000000000000000"), // 5
+  BigNumber.from("10000000000000000000"), // 10
+  BigNumber.from("15000000000000000000"), // 15
+  BigNumber.from("20000000000000000000"), // 20
+];
+const testActiveBalanceCurrentEpoch: BigNumber[] = [
+  BigNumber.from("1000000000000000000"), // 1
+  BigNumber.from("9000000000000000000"), // 9
+  BigNumber.from("15000000000000000000"), // 15
+  BigNumber.from("22000000000000000000"), // 22
+];
+
+describe("Insolvent Contract test suite", () => {
+  const mockProvider: MockEthersProvider = new MockEthersProvider();
```

Comment:
I suggest adding a test case where different blocks are being handled by the bot and multiple findings are generated, since all test cases are with a single block.